webSocket/transceiver/transceiver.go

```diff
@@ -124,6 +138,25 @@ func (wt *wsTransceiver) verifyPayloadAndSendAckIfNeeded(connection webSocket.WS
 	wt.sendAckIfNeeded(connection, payloadData)
 }
 
+func (wt *wsTransceiver) handleAckMessage(payload []byte) {
+	counter, err := wt.payloadParser.ExtractUint64FromAckMessage(payload)
+	if err != nil {
+		wt.log.Debug("wsTransceiver.handleAckMessage cannot decode the ack message", "error", err)
+		return
+	}
+
+	wt.log.Trace("wt.handleAckMessage: received ack", "counter", counter)
+	expectedCounter := atomic.LoadUint64(&wt.counter)
+	if expectedCounter == counter {
+		wt.ackChan <- struct{}{}
```

Comment:
select with default? What happens if a go routine ends up here but the `waitForAck` function will always take the `case <-wt.safeCloser.ChanClose():` case ?